modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/SerializableSerializer.cs

```diff
@@ -603,13 +597,33 @@ private static object ReadAsCustomType(Type customType, SerializationInfo serInf
         {
             var ctorFunc = SerializableTypeDescriptor.Get(customType).SerializationCtor;
 
-            var customObj = ctorFunc(serInfo, ctx);
+            var customObj = ctorFunc != null
+                ? ctorFunc(serInfo, ctx)
+                : FormatterServices.GetUninitializedObject(customType);
 
             var wrapper = customObj as IObjectReference;
 
-            return wrapper == null
+            var resObj = wrapper == null
                 ? customObj
                 : wrapper.GetRealObject(ctx);
+
+            // Special case: type is replaced, but there is no serialization ctor.
+            // Example: StringComparer.OrdinalIgnoreCase.
+            if (ctorFunc == null)
+            {
+                // Cached internally.
+                var members = FormatterServices.GetSerializableMembers(resObj.GetType());
```

Comment:
What about non-serializable members? What if non-serializable method greatly affects object functionality?

Reply:
`SerializableSerializer` handles `ISerializable` types only. If a member of such type is marked with `[NonSerialized]`, it is skipped during serialization. This is what `GetSerializableMembers` does - returns all members that do not have `[NonSerialized]` attribute.

We are just following the same logic as the built-in binary serializer has.

Reply:
I see now